src/vector_topic_modeling/evaluation.py

```
@@ -0,0 +1,227 @@
+    if not vectors:
+        return []
+    dims = len(vectors[0])
+    centroid = [0.0] * dims
+    for v in vectors:
+        for i in range(dims):
+            centroid[i] += v[i]
+    count = len(vectors)
+    return [x / count for x in centroid]
+
+
+def calculate_extended_metrics(
```

Ruff (0.15.7)

[warning] Too many branches (20 > 12)

(PLR0912)